### PR TITLE
Add libsmb2 

### PR DESCRIPTION
**Pull request by Wolf3s**

### CODE REVIEW

**Comment by Wolf3s** [comment]
Requires: https://github.com/sahlberg/libsmb2/pull/345

File: build-cmakelibs.sh [hunk 1]
```diff
@@ -88,9 +102,7 @@ git clone --depth 1 -b release-2.6.3 https://github.com/libsdl-org/SDL_mixer.git
 git clone --depth 1 -b release-2.6.3 https://github.com/libsdl-org/SDL_image.git || { exit 1; }
 git clone --depth 1 -b release-2.20.2 https://github.com/libsdl-org/SDL_ttf.git || { exit 1; }
 
-# We need to clone the whole repo and point to the specific hash for now,
-# till a new version is released after this commit
-
```
Comment (fjtrujy) [inline build-cmakelibs.sh:93]: Don’t remove this comments…

File: build-cmakelibs.sh [hunk 2]
```diff
@@ -88,9 +102,7 @@ git clone --depth 1 -b release-2.6.3 https://github.com/libsdl-org/SDL_mixer.git
...
+git clone --depth 1 https://github.com/sahlberg/libsmb2.git || { exit 1; }
```
Comment (fjtrujy) [inline build-cmakelibs.sh:105]: You need to checkout a concrete hash, take a look how others do (like SDL2)

Comment (Wolf3s) [reply build-cmakelibs.sh:105]: First of all i´m waiting @sahlberg merge this pull request [#](https://github.com/sahlberg/libsmb2/pull/352) then i will open this as soon it´s merged.

Comment (fjtrujy) [reply build-cmakelibs.sh:105]: It has been merged already

File: build-cmakelibs.sh [hunk 3]
```diff
@@ -139,5 +159,18 @@ install -m644 argtable2-13/src/argtable2.h $PS2SDK/ports/include/
...
+cp -fr $PS2SDK/ports/lib/libsmb2_ip.a $PS2SDK/ports/lib/libsmb2.a
+CFLAGS="-DPS2IPS" build libsmb2
+cp -fr $PS2SDK/ports/lib/libsmb2_ip.a $PS2SDK/ports/lib/libsmb2_ips.a
+cp -fr $PS2SDK/ports/lib/libsmb2.a $PS2SDK/ports/lib/libsmb2_ip.a
+rm -fr $PS2SDK/ports/lib/libsmb2.a
+build_iop libsmb2
+build enet
+
+CFLAGS="-Wno-implicit-function-declaration" build argtable2-13 -DHAVE_STRINGS_H=ON -DHAVE_STDC_HEADERS=ON
+# Copy manually the argtable2.h header
```
Comment (fjtrujy) [inline build-cmakelibs.sh:172]: Let arguable 2 and 3 together, as it was before and add your changes at the end

File: build-cmakelibs.sh [hunk 4]
```diff
@@ -139,5 +156,13 @@ install -m644 argtable2-13/src/argtable2.h $PS2SDK/ports/include/
 
 build argtable3 -DARGTABLE3_INSTALL_CMAKEDIR="${PS2SDK}/ports/lib/cmake/" -DARGTABLE3_REPLACE_GETOPT=OFF -DARGTABLE3_ENABLE_EXAMPLES=OFF -DARGTABLE3_ENABLE_TESTS=OFF
 
+build libsmb2 
+cp -fr $PS2SDK/ports/lib/libsmb2_ip.a $PS2SDK/ports/lib/libsmb2.a
```
Comment (fjtrujy) [inline build-cmakelibs.sh:160]: Final final final thing... let's fix the static analysis error:
`cp -fr "${PS2SDK}/ports/lib/libsmb2_ip.a" "${PS2SDK}/ports/lib/libsmb2_ips.a"`

And do the same approach with the following lines

**Pull request merged**